fence/README.md

```diff
@@ -0,0 +1,151 @@
+| SP       | Storage Provider, the vendor of a CSI plugin implementation.                          |
+| DR       | Disaster Recovery.                                                                    |
+| RPC      | [Remote Procedure Call](https://en.wikipedia.org/wiki/Remote_procedure_call).         |
+
+## Objective
+
+Define a standard that will enable storage vendors (SP) to
+perform node level fencing using corresponding CIDR blocks.
+
+### Goals in MVP
+
+The new extension will define a procedure that allows SP to block
+access to any client using corresponding CIDR blocks, and unblock
+as per the requirement.
+
+If multiple COs are accessing a shared resource provided by the SP, and a node
+failure takes place, this procedure will allow the SP to block access from any
+node of CO, and thus preventing further access and corruption of data.
+
+### Non-Goals in MVP
+
+* Application level fencing, i.e blocklisting by maintaining a list of applications
+  that are to be denied access.
+
+## Solution Overview
+
+This specification defines an interface along with the minimum operational and
+packaging recommendations for a storage provider (SP) to implement fencing operation.
+The interface declares the RPCs that a plugin MUST expose.
+
+## RPC Interface
+
+* **Controller Service**: The Controller plugin MUST implement these sets of
+  RPCs.
+
+```protobuf
+syntax = "proto3";
+package fence;
+
+import "google/protobuf/descriptor.proto";
+import "github.com/container-storage-interface/spec/lib/go/csi/csi.proto";
+
+option go_package = ".;fence";
+
+// Controller holds the RPC method for performing fencing operations.
+service Controller {
+  // FenceClusterNetwork RPC call to perform a fencing operation.
+  rpc FenceClusterNetwork (FenceClusterNetworkRequest)
+  returns (FenceClusterNetworkResponse) {}
+
+  // UnfenceClusterNetwork RPC call to remove a list of CIDR blocks from the
+  // list of blocklisted/fenced clients.
+  rpc UnfenceClusterNetwork (UnfenceClusterNetworkRequest)
+  returns (UnfenceClusterNetworkResponse) {}
+
+  // ListClusterFence RPC call to provide a list of blocklisted/fenced clients.
+  rpc ListClusterFence(ListClusterFenceRequest)
+  returns (ListClusterFenceResponse){}
+}
+```
+
+### FenceClusterNetwork
+
+```protobuf
+// FenceClusterNetworkRequest contains the information needed to identify
+// the cluster so that the appropriate fencing operation can be
+// performed.
+message FenceClusterNetworkRequest {
+  // Plugin specific parameters passed in as opaque key-value pairs.
+  map<string, string> parameters = 1;
+  // Secrets required by the plugin to complete the request.
+  map<string, string> secrets = 2 [(csi.v1.csi_secret) = true];
+  // list of CIDR blocks on which the fencing operation is expected to be
+  // performed.
+  repeated CIDR cidrs = 3;
+}
+
+// FenceClusterNetworkResponse holds the information about the result of the
```

Comment:
`holds the information about the result` is not correct anymore. Maybe mention that the contents of the response is intentionally left empty (see `GetIdentity` in `identity/README.md`)?